pallets/grants/src/lib.rs

```diff
@@ -214,6 +214,48 @@ pub mod pallet {
 
 			Ok(().into())
 		}
+
+		/// Initiate a bridge transfer of all vested funds to the given `eth_address` on the given `chain_id`.
+		/// This process will need to be completed by the bridge oracles.
+		#[pallet::call_index(4)]
+		#[pallet::weight(T::WeightInfo::bridge_all_vesting_schedules())]
+		pub fn bridge_all_vesting_schedules(
+			origin: OriginFor<T>,
+			eth_address: [u8; 20],
+			chain_id: u64,
+		) -> DispatchResultWithPostInfo {
+			let from = ensure_signed(origin)?;
+
+			let locked_amount_left = Self::do_claim(&from);
+			if locked_amount_left.is_zero() {
+				<VestingSchedules<T>>::remove(from);
+				Self::deposit_event(Event::NoVestedFundsToBridgeAfterClaim);
+				return Ok(().into());
+			}
+
+			let free_balance = T::Currency::free_balance(&from);
+			let bridgeable_funds = locked_amount_left.min(free_balance);
```

Comment:
I am not sure why this is necessary, can you clarify? I assume you are trying to cover some edge case?

Reply:
It's exactly equal to cancel_all have a look at that. Yes it's covering a corner case

Reply:
Let's have a look at https://github.com/NodleCode/chain/pull/624

Reply:
It's a historical thing. In the past some of the vesting calls where not safe and they could cause corrupt state. In those corrupt state, the user could end up having less free coins that was granted to them. This `minimum` check ensures the call will succeed safely even for those failures.